examples/applications/plot_time_series_lagged_features.py

```diff
@@ -258,36 +268,30 @@ def consolidate_scores(cv_results, scores, metric):
             metric = key.split("test_")[1]
             scores = consolidate_scores(cv_results, scores, metric)
 
-df = pd.DataFrame(scores)
-
-styled_df_copy = df.copy()
-
+styled_df = pl.DataFrame(scores)
 
-def extract_numeric(value):
-    parts = value.split("±")
-    mean_value = float(parts[0])
-    std_value = float(parts[1].split()[0])
 
-    return mean_value, std_value
+def extract_numeric(col):
+    return pl.col(col).str.split(by=" ").list.gather([0, 2]).cast(pl.List(pl.Float64))
 
 
-# Convert columns containing "±" to tuples of numerical values
-cols_to_convert = df.columns[1:]  # Exclude the "loss" column
-for col in cols_to_convert:
-    df[col] = df[col].apply(extract_numeric)
+def list_min(col):
+    # Polars doesn't currently support finding a minimum of a list
+    # so we find the minimum by sorting.
+    return col.sort_by([col.list.first(), col.list.last()]).first()
 
-min_values = df.min()
 
-# Create a mask for highlighting minimum values
-mask = pd.DataFrame("", index=df.index, columns=df.columns)
-for col in cols_to_convert:
-    mask[col] = df[col].apply(
-        lambda x: "font-weight: bold" if x == min_values[col] else ""
+gt_styled_df = GT(styled_df)
+(
+    gt_styled_df.tab_style(
+        style.text(weight="bold"),
+        [
+            loc.body(col, extract_numeric(col) == list_min(extract_numeric(col)))
+            for col in styled_df.columns
+            if col != "loss"
+        ],

```

Comment:
Replacing pandas styling with great tables as they have native support for Polars. Apart from `fetch_openml` this completely replaces pandas.